evalml/tests/automl_tests/test_search.py

```diff
@@ -47,6 +48,54 @@ def test_search_data_check_error(
     pd.testing.assert_series_equal(target, infer_feature_types(y))
 
 
+@pytest.mark.parametrize(
+    "problem_config", [None, "missing_date_index", "has_date_index"]
+)
+def test_search_data_check_error_timeseries(problem_config):
+    X, y = pd.DataFrame({"features": range(30)}), pd.Series(range(30))
+    problem_configuration = None
+
+    dates = pd.date_range("2021-01-01", periods=30)
```

Comment:
Wondering if there's a better way to do this, since it's done a few times. Maybe something along the lines of:
`dates = pd.date_range("2021-01-01", periods=29).append(pd.date_range("2021-01-31", periods=1))` would accomplish the same without having to convert to list, change, and convert back to `DatetimeIndex`?